### PR TITLE
command: Remove executable parameter when using command

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY

This was discussed with the core team and removing this option was preferred.
For backward compatibility we accept the parameter, but warn the user instead.

This PR also includes a change related to warnings, we now use the warning-infrastructure, this avoids empty warnings-dictionary in the output if there's no warning.

##### ISSUE TYPE

 - Bugfix Pull Request
 - Docs Pull Request

##### COMPONENT NAME

command

##### ANSIBLE VERSION

v2.4

### CODE REVIEW

**Comment by dagwieers** [comment]
@nitzmahone  @jimi-c @bcoca Is this implementation acceptable ?

**Review by nitzmahone — APPROVED** [review]
Looks good to me- I don't think returning an empty warnings collection is a problem, but others might disagree.

**Comment by dagwieers** [comment]
@nitzmahone The empty warnings was not a concern to me, I simply wanted to make use of the new module.warn() infrastructure. More examples of it in use may help people do the right thing :-)

@bcoca So what needs to happen to get this going ?

File: lib/ansible/modules/commands/command.py [hunk 1]
```diff
@@ -157,6 +141,10 @@ def main():
     removes = module.params['removes']
     warn = module.params['warn']
 
+    if not shell and executable:
+        module.warn('The parameter "executable" is not supported with the "command" module. Not using "%s".' % executable)
```
Comment (abadger) [inline lib/ansible/modules/commands/command.py]: Maybe add the version executable went away:
``` python
module.warn('As of Ansible 2.4, the parameter "executable" is no longer supported with the "command" module. Not using "%s".' % executable) 
```

Comment (dagwieers) [reply lib/ansible/modules/commands/command.py]: Changed. Thanks !

**Comment by abadger** [comment]
merged to devel.  Thanks dag!

**Pull request merged**